### PR TITLE
Add config-file option to get nodegroup

### PR DESCRIPTION
**Pull request by Skarlso**
### Description

Does the `nodegroup` part of https://github.com/weaveworks/eksctl/issues/1126.

```
eksctl get nodegroup -f cluster-managed.yaml
2021-11-16 16:03:36 [ℹ]  eksctl version 0.75.0-dev+c5f6d0ef.2021-11-16T15:10:41Z
2021-11-16 16:03:36 [ℹ]  using region us-west-2
CLUSTER                         NODEGROUP       STATUS  CREATED                 MIN SIZE        MAX SIZE        DESIRED CAPACITY     INSTANCE TYPE    IMAGE ID        ASG NAME
test-nodegroup-cluster-config   managed-ng-1    ACTIVE  2021-11-16T12:02:30Z    1               2               1                    m5.large AL2_x86_64      eks-52be9450-42a1-0ff9-b901-3b51d3bda8d3
test-nodegroup-cluster-config   managed-ng-2    ACTIVE  2021-11-16T14:56:25Z    1               2               1                    m5.large AL2_x86_64      eks-f8be949f-df08-fa4e-4e46-2590f597f474
```

It's not allowed to define `--name` with `--config-file`. I'm on the fence with that. Like, it would be nice to define which nodegroup you are interested in... But then again, if you use the config-file you want to see all of the things the config-file provides and this behaviour is mimicked with others. 

This was easier since nodegroup already was using the config thing to load in data, so it was easy to just overwrite it if the setting is set. I tried without, with and only name and only name argument.

### Testing

- [x] with `--config-file`
- [x] with named argument
- [x] with `--cluster` + `--name`
- [x] with `--config-file` + `--name`
- [x] without anything

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/ctl/get/nodegroup.go [hunk 1]
```diff
@@ -38,15 +38,18 @@ func getNodeGroupCmd(cmd *cmdutils.Cmd) {
...
 		cmdutils.AddCommonFlagsForGetCmd(fs, &params.chunkSize, &params.output)
 		cmdutils.AddTimeoutFlag(fs, &cmd.ProviderConfig.WaitTimeout)
+		cmdutils.AddConfigFileFlag(fs, &cmd.ClusterConfigFile)
 	})
 
 	cmdutils.AddCommonFlagsForAWS(cmd.FlagSetGroup, &cmd.ProviderConfig, false)
 }
 
 func doGetNodeGroup(cmd *cmdutils.Cmd, ng *api.NodeGroup, params *getCmdParams) error {
+	if err := cmdutils.NewMetadataLoader(cmd).Load(); err != nil {
```
Comment (Skarlso) [inline pkg/ctl/get/nodegroup.go:48]: Oh, I failed to write some unit tests for this 🤦

File: pkg/ctl/get/nodegroup.go [hunk 2]
```diff
@@ -38,15 +38,18 @@ func getNodeGroupCmd(cmd *cmdutils.Cmd) {
...
+		return err
+	}
 	cfg := cmd.ClusterConfig
 
-	// TODO: move this into a loader when --config-file gets added to this command
```
Comment (nikimanoledaki) [inline pkg/ctl/get/nodegroup.go:49]: I think this TODO is saying to move all the validation below into the loader to match the pattern in other `ctl`s 🤔

Comment (Skarlso) [reply pkg/ctl/get/nodegroup.go:49]: Oh... you might be right. :D

File: pkg/ctl/get/nodegroup_test.go [hunk 3]
```diff
@@ -15,12 +17,41 @@ var _ = Describe("get", func() {
...
+
+		It("setting --name and --config-file at the same time", func() {
+			f, err := os.CreateTemp("", "configfile")
+			Expect(err).NotTo(HaveOccurred())
+			_, err = f.WriteString(configFile)
+			Expect(err).NotTo(HaveOccurred())
+			cmd := newMockCmd("nodegroup", "--name", "name", "--config-file", f.Name())
+			_, err = cmd.execute()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("Error: cannot use --name when --config-file/-f is set"))
```
Comment (cPu1) [inline pkg/ctl/get/nodegroup_test.go:41]: There's also `MatchError`:

```suggestion
			Expect(err).To(MatchError(ContainSubstring("Error: cannot use --name when --config-file/-f is set")))
```

File: pkg/ctl/get/nodegroup_test.go [hunk 4]
```diff
@@ -15,12 +17,41 @@ var _ = Describe("get", func() {
...
+
+		It("setting --cluster and --config-file at the same time", func() {
+			f, err := os.CreateTemp("", "configfile")
+			Expect(err).NotTo(HaveOccurred())
+			_, err = f.WriteString(configFile)
+			Expect(err).NotTo(HaveOccurred())
+			cmd := newMockCmd("nodegroup", "--cluster", "name", "--config-file", f.Name())
+			_, err = cmd.execute()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("Error: cannot use --cluster when --config-file/-f is set"))
```
Comment (cPu1) [inline pkg/ctl/get/nodegroup_test.go:52]: Same as the previous comment about using `MatchError`.

File: pkg/ctl/get/nodegroup_test.go [hunk 5]
```diff
@@ -15,12 +17,41 @@ var _ = Describe("get", func() {
 		})
 
 		It("setting --name and argument at the same time", func() {
-			cmd := newMockCmd("nodegroup", "ng", "--cluster", "dummy", "--name", "ng")
```
Comment (cPu1) [inline pkg/ctl/get/nodegroup_test.go:18]: Any reason why the `--cluster` argument was removed from this test?

Comment (Skarlso) [reply pkg/ctl/get/nodegroup_test.go:18]: This one is a bit of a interesting one. I think it was testing the wrong thing. However, with the introduction of the new `--config-file` option the error message is not correctly returning that `--name and nameArg can't be together, where as previously, it was doing the wrong thing with the `--cluster` param. Very weird. :D

**Review by cPu1 — APPROVED** [review]

File: pkg/ctl/get/nodegroup.go (hunk 2 above)
Comment (nikimanoledaki) [reply pkg/ctl/get/nodegroup.go:49]: Ah I meant that this could be done as part of this PR! WDYT?

**Comment by nikimanoledaki** [comment]
`It's not allowed to define --name with --config-file`
This reminds me of https://github.com/weaveworks/eksctl/pull/4383 where, if `--config-file` and `--name` are both set, `FindNodegroup` discovers the nodegroup in question in the config file. WDYT? Maybe worth creating a new issue for this feature though!

File: pkg/ctl/get/nodegroup.go (hunk 2 above)
Comment (Skarlso) [reply pkg/ctl/get/nodegroup.go:49]: definitely not :DD

**Comment by Skarlso** [comment]
@nikimanoledaki yeah sure. The problem is that it's actually in the main thing and it would change behaviour for all commands. Which might be fine of course. Just need to document it ( if we do ) and be aware.

File: pkg/ctl/get/nodegroup.go (hunk 2 above)
Comment (aclevername) [reply pkg/ctl/get/nodegroup.go:49]: why not? Seems related to this PR :thinking:

**Comment by nikimanoledaki** [comment]
That's true, it would be a big change in behaviour that would be hard to justify if there hasn't been any demand from users. The difference with the PR I linked is that there had been user demand! So I guess let's leave it as is and let user demand drive this, like all other things? :)

**Comment by Skarlso** [comment]
Yep, sounds good to me.

File: pkg/ctl/get/nodegroup_test.go (hunk 5 above)
Comment (Skarlso) [reply pkg/ctl/get/nodegroup_test.go:18]: Uh, I just understood what's happening here and will provide a fix in an upcoming PR.

**Pull request merged**